playground/backend/internal/setuper/setuper.go

```diff
@@ -0,0 +1,97 @@
+package setuper
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api/v1"
+	"beam.apache.org/playground/backend/internal/environment"
+	"beam.apache.org/playground/backend/internal/executors"
+	"beam.apache.org/playground/backend/internal/fs_tool"
+	"beam.apache.org/playground/backend/internal/logger"
+	"beam.apache.org/playground/backend/internal/validators"
+	"fmt"
+	"github.com/google/uuid"
+)
+
+// SetupLifeCycle creates fs_tool.LifeCycle and prepares files and folders needed to code processing
+func SetupLifeCycle(sdk pb.Sdk, code string, pipelineId uuid.UUID, workingDir string) (*fs_tool.LifeCycle, error) {
+	// create file system service
+	lc, err := fs_tool.NewLifeCycle(sdk, pipelineId, workingDir)
+	if err != nil {
+		logger.Errorf("%s: RunCode(): NewLifeCycle(): %s\n", pipelineId, err.Error())
+		return nil, err
+	}
+
+	// create folders
+	err = lc.CreateFolders()
+	if err != nil {
+		logger.Errorf("%s: RunCode(): CreateFolders(): %s\n", pipelineId, err.Error())
+		return nil, err
+	}
+
+	// create file with code
+	_, err = lc.CreateExecutableFile(code)
+	if err != nil {
+		logger.Errorf("%s: RunCode(): CreateExecutableFile(): %s\n", pipelineId, err.Error())
+		return nil, err
+	}
+	return lc, nil
+}
+
+// SetupCompileBuilder returns executors.CompileBuilder with validators and compiler based on sdk
+func SetupCompileBuilder(lc *fs_tool.LifeCycle, sdk pb.Sdk, executorConfig *environment.ExecutorConfig) (*executors.CompileBuilder, error) {
+	filePath := lc.GetAbsoluteExecutableFilePath()
+	val, err := SetupValidators(sdk, filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	compileBuilder := executors.NewExecutorBuilder().
+		WithValidator().
+		WithSdkValidators(val).
+		WithCompiler()
+
+	workingDir := lc.GetAbsoluteExecutableFilesFolderPath()
+
+	compileBuilder = compileBuilder.
+		WithCommand(executorConfig.CompileCmd).
+		WithArgs(executorConfig.CompileArgs).
+		WithFileName(filePath).
+		WithWorkingDir(workingDir)
+
+	return compileBuilder, nil
+}
+
+// SetupRunBuilder returns executors.RunBuilder based on sdk
+func SetupRunBuilder(pipelineId uuid.UUID, lc *fs_tool.LifeCycle, sdk pb.Sdk, env *environment.Environment, compileBuilder *executors.CompileBuilder) (*executors.RunBuilder, error) {
+	runBuilder := compileBuilder.
+		WithRunner().
+		WithCommand(env.BeamSdkEnvs.ExecutorConfig.RunCmd).
+		WithArgs(env.BeamSdkEnvs.ExecutorConfig.RunArgs).
+		WithWorkingDir(lc.GetAbsoluteExecutableFilesFolderPath())
+
+	switch sdk {
+	case pb.Sdk_SDK_JAVA:
+		className, err := lc.ExecutableName(pipelineId, env.ApplicationEnvs.WorkingDir())
+		if err != nil {
+			logger.Errorf("%s: get executable file name: %s\n", pipelineId, err.Error())
+			return nil, err
+		}
+
+		runBuilder = runBuilder.
+			WithClassName(className)
+	default:
+		return nil, fmt.Errorf("incorrect sdk: %s", sdk)
```

Comment:
Should we use here one of our errors or not?

Reply:
I guess not. We use `our` errors in case we send an error to the client.

Reply:
seems like it's not our logger, why?

Reply:
Yes, that's true. We don't need to identify it in the call stack because it's just for log purposes. We don't have any scenario where we should process this error. lgtm